### ISSUE TITLE
generator interpreted as partially unknown inside sum

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#5183]
**Describe the bug**
The following code causes an error

**To Reproduce**
```python
# pyright: strict

from __future__ import annotations


def f(data: dict[str, int]):
    # this is information: Type of "data.get(k, 0) for k in "ABC"" is "Generator[int, None, None]"
    reveal_type(data.get(k, 0) for k in "ABC")
    # but this logs
    # error: Argument type is partially unknown
    # Argument corresponds to parameter "__iterable" in function "sum"
    # Argument type is "Generator[int | Unknown, None, None]" (reportUnknownArgumentType)
    sum(data.get(k, 0) for k in "ABC")
```

**Expected behavior**
should not cause any error

**VS Code extension or command-line**
pyright 1.1.310

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.311, which I just published. It will also be included in a future release of pylance.
